### PR TITLE
New navigation region dropdown

### PR DESCRIPTION
**Pull request by pflopez**
PLAN-325
PLAN-333

- Adds a feature flag for new navigation.  (`new_navigation`)
- Adds functionality to the feature flag directive to hide content based on a feature flag
- Moves region dropdown to component
- Hides region dropdown on top nav for users under the flag
- Shows region dropdown on map controls for users under the flag
- Shows feedback btn on top nav for users under the flag
- Hides help icon for users under the flag. The help icon is going to live on a secondary navigation bar (not included on this pr)

With the new navigation flag turned on (so far): 
Screen Shot 2023-08-28 at 11 55 18

Testing:
- Users not on the feature flag shouldn't see any changes
- Users under the `new_navigation` flag should not see the help icon at the top, and should see the region dropdown under the map controls.

### CODE REVIEW

File: src/interface/src/app/features/README.md [hunk 1]
```diff
@@ -48,6 +48,9 @@ This flag will enable the Project Area control panel in the Scenario Configurati
 This flag will enable use of ITS data for past resilience projects rather than
 Calmapper.
 
+### new_navigation
+This flag controls the changes regarding the new navigation
```
Comment (runforyourlives) [inline src/interface/src/app/features/README.md:52]: Can you describe what the new navigation is?  (e.g. top bar, etc.)

File: src/interface/src/app/features/feature.guard.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { inject } from '@angular/core';
 import { CanActivateFn, CanMatchFn, Router } from '@angular/router';
 
-import { FeatureService } from './feature.service';
+import { FeatureService} from './feature.service';
```
Comment (sachiyuan) [inline src/interface/src/app/features/feature.guard.ts:4]: nit: spacing (to make consistent with other imports { FeatureService } vs { FeatureService})

File: src/interface/src/app/region-dropdown/region-dropdown.component.scss [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    border-top: 5px solid #3367D6;
+    pointer-events: none;
+  }
+}
+
+.region {
+  &::-ms-expand {
+    display: none; /* IE */
+  }
+
```
[the remark this answers is not in the document]
Comment (pflopez) [reply src/interface/src/app/region-dropdown/region-dropdown.component.scss:86]: good catch! thanks

**Review by runforyourlives — COMMENTED** [review]
My main recommendation is to hide the help icon once you have the secondary bar available (in a subsequent PR), rather than in this CL.  There's always a chance this gets demo'ed on demo.planscape.org, and that would yield an experience with no help widget.

**Comment by pflopez** [comment]
@runforyourlives thanks for the recommendation, however the icon is hidden only when the feature flag is on, so it should be  visible on demo.planscape.com

**Review by runforyourlives — APPROVED** [review]

**Pull request merged**